lms/djangoapps/lti_provider/users.py

```diff
@@ -51,13 +60,32 @@ def create_lti_user(lti_user_id, lti_consumer):
     created = False
     while not created:
         try:
-            edx_user_id = generate_random_edx_username()
-            edx_email = "{}@{}".format(edx_user_id, settings.LTI_USER_EMAIL_DOMAIN)
+            if lti_email is not None:
+                edx_email=cut_to_max_len(lti_email, EMAIL_SIZE)
+                edx_username=cut_to_max_len(lti_email, USERNAME_SIZE)
+                try:
+                    #if this username already exists then use the random username
+                    edx_user = User.objects.get(username=edx_username)
+                    edx_username=generate_random_edx_username()
+                except User.DoesNotExist:
+                    pass
+            else:
+                edx_username=generate_random_edx_username()
+                edx_email = "{}@{}".format(edx_username, settings.LTI_USER_EMAIL_DOMAIN)
+
             edx_user = User.objects.create_user(
-                username=edx_user_id,
+                username=edx_username,
                 password=edx_password,
                 email=edx_email,
             )
+
+            if edx_user is not None:
```

Comment:
Can we set these when we create the User above rather than following with an update?